src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.Generated.cs

```diff
@@ -369,137 +371,137 @@ private void ExtraFeatureSet(Type key, object value)
+                feature = (TFeature?)_currentIHttpResponseTrailersFeature;
             }
             else if (typeof(TFeature) == typeof(IResponseCookiesFeature))
             {
-                feature = (TFeature)_currentIResponseCookiesFeature;
+                feature = (TFeature?)_currentIResponseCookiesFeature;
             }
             else if (typeof(TFeature) == typeof(IItemsFeature))
             {
-                feature = (TFeature)_currentIItemsFeature;
+                feature = (TFeature?)_currentIItemsFeature;
             }
             else if (typeof(TFeature) == typeof(ITlsConnectionFeature))
             {
-                feature = (TFeature)_currentITlsConnectionFeature;
+                feature = (TFeature?)_currentITlsConnectionFeature;
             }
             else if (typeof(TFeature) == typeof(IHttpWebSocketFeature))
             {
-                feature = (TFeature)_currentIHttpWebSocketFeature;
+                feature = (TFeature?)_currentIHttpWebSocketFeature;
             }
             else if (typeof(TFeature) == typeof(ISessionFeature))
             {
-                feature = (TFeature)_currentISessionFeature;
+                feature = (TFeature?)_currentISessionFeature;
             }
             else if (typeof(TFeature) == typeof(IHttpMaxRequestBodySizeFeature))
             {
-                feature = (TFeature)_currentIHttpMaxRequestBodySizeFeature;
+                feature = (TFeature?)_currentIHttpMaxRequestBodySizeFeature;
             }
             else if (typeof(TFeature) == typeof(IHttpMinRequestBodyDataRateFeature))
             {
-                feature = (TFeature)_currentIHttpMinRequestBodyDataRateFeature;
+                feature = (TFeature?)_currentIHttpMinRequestBodyDataRateFeature;
             }
             else if (typeof(TFeature) == typeof(IHttpMinResponseDataRateFeature))
             {
-                feature = (TFeature)_currentIHttpMinResponseDataRateFeature;
+                feature = (TFeature?)_currentIHttpMinResponseDataRateFeature;
             }
             else if (typeof(TFeature) == typeof(IHttpBodyControlFeature))
             {
-                feature = (TFeature)_currentIHttpBodyControlFeature;
+                feature = (TFeature?)_currentIHttpBodyControlFeature;
             }
             else if (typeof(TFeature) == typeof(IHttpResetFeature))
             {
-                feature = (TFeature)_currentIHttpResetFeature;
+                feature = (TFeature?)_currentIHttpResetFeature;
             }
             else if (MaybeExtra != null)
             {
-                feature = (TFeature)(ExtraFeatureGet(typeof(TFeature)));
+                feature = (TFeature?)(ExtraFeatureGet(typeof(TFeature)));
             }
 
             if (feature == null)
             {
                 feature = ConnectionFeatures.Get<TFeature>();
             }
 
-            return feature;
+            return feature!;
```

Comment:
Should `IFeatureCollection.Get<T>()` return a nullable value? It's technically correct, but could harm usability of the API.

@pranavkm is there a reason it was left as non-null in https://github.com/dotnet/aspnetcore/pull/22337?

Reply:
I think it's intended to return null if the feature does not exist (`ISessionFeature`, `IEndpointFeature` etc are optional). What would be useful is to introduce a `GetRequiredFeature` for cases like this where we expect the feature to exist.